internal/mocktc/queue.go

```diff
@@ -167,6 +167,14 @@ func (queue *Queue) CreateArtifact(taskId, runId, name string, payload *tcqueue.
 	return &par, nil
 }
 
+func (queue *Queue) FinishArtifact(taskId, runId, name string, payload *tcqueue.FinishArtifactRequest) error {
+	queue.mu.Lock()
+	defer queue.mu.Unlock()
+
+	// do nothing (for now)
+	return nil
+}
```

Comment:
Should we also update `CreateArtifact` method above, and add handling of `"object"`storage type? At the moment it will cause a test failure - so I guess we don't have any tests for object storage type - should we add at least one?

Reply:
It [is tested](https://github.com/taskcluster/taskcluster/blob/51ac97dcac93f8a219a464c0753a8d914899a449/workers/generic-worker/artifacts_test.go#L119-L147) like the other upload methods.  But it looks like that only tests creating the TaskArtifact objects, without actually executing the upload.

Ah, I found tests that actually create objects in the broken-on-docker file!

Reply:
Oh indeed, that isn't named nicely (broken-on-docker). I think we'll probably get rid of the build constraints entirely next year.